### ISSUE TITLE
Mac: Builds for one app attempting to parse styles for all

### ISSUE THREAD

**Issue by mcblum** [nrwl/nx#288]
We have a weird problem that just popped up once we added a second app to the workspace. Both apps build successfully using `jit` but when building with `aot`, building either app fails because it tries to parse the styles of both apps and there are some mixins that are in one app but not the other. We currently have no shared components so it's not that it's trying to use the same component.

Here's an example of what we're running:

`ng build --aot --app=app_1`

Which fails with:

```
ERROR in Error: Child compilation failed:
Module build failed: 
  height: $headerHeight;
         ^
      Undefined variable: "$headerHeight".
      in /www/monorepo/apps/app_2/src/app/view-components/header/header.component.scss (line 8, column 11):
Error: 
```

The weird thing is that `app_2` isn't involved in any way with `app_1`, and yet it's still trying to build it.

If anyone has any ideas let me know! This one has stumped me all day.

**Comment by thelgevold** [comment]
Hi, do you have a small repro to reproduce the problem?

**Comment by mcblum** [comment]
@thelgevold I'll try to get one online somewhere for you to clone and run. If you have any repo yourself that uses Nx you should also be able to replicate it by doing something like:

```
 "styles": [
        "styles.scss"
      ],
      "stylePreprocessorOptions": {
        "includePaths": [
          "sass"
        ]
      },
```

and then put a file in {appRoot}/sass/_mixins.scss. If you have two apps that both use an include path, even if that path is different, building one app will scan all apps meaning that if `_mixins.scss` doesn't have exactly the same mixin names it will fail. It's not actually a problem once built, you only get the relevant CSS for the app, but during the parsing stage it will look at all apps.

**Comment by ThomasBurleson** [comment]
@mcblum - please provide a sample workspace (repo) that duplicates this issue; otherwise we will need to close.

**Comment by mcblum** [comment]
@ThomasBurleson hey could you give me a few more days? We got really unlucky here and had a sick kid and then they canceled school on us today and tomorrow. I haven't forgotten just haven't quite gotten there yet.

**Comment by thelgevold** [comment]
This may also be an issue with Angular CLI since nx inherits the apps feature from there. Does this work in a plain Angulat Cli project?

**Comment by ThomasBurleson** [comment]
@mcblum - np, thanks for the response.

**Comment by mcblum** [comment]
Hey. An interesting thing just happened - I went to reproduce this issue with a new workspace and I can't do it. We started with Nx right when it was released, and have kept upgrading as we went along, but maybe there's something with the config that is causing our instance to break? Is there an easy way to reset everything?

**Comment by mcblum** [comment]
Woah - I figured it out. When I created the new app, it included this line `"tsconfig": "tsconfig.app.json",` which was different from what we had. Our app pointed to tsconfig at the root, whereas the new Nx Workspace had one tsconfig for each app. Since all of our apps were pointing to the root, when building for AOT the compiler was checking everything, not just the app in question.

This is a very good day and hopefully if someone else has this issue they'll find this thread. Thanks guys! Appreciate the hard work.
